s3stream/src/main/java/com/automq/stream/FixedSizeByteBufPool.java

```diff
@@ -67,6 +67,10 @@ private ByteBuf allocate() {
     public void release(ByteBuf buffer) {
         assert buffer.capacity() == bufferSize;
 
+        if (buffer.refCnt() != 1) {
+            throw new IllegalArgumentException("The buffer is not exclusive");
+        }
+
```

Comment:
This is a high-frequency method call, so for performance reasons, I think using `assert` here would be sufficient. If a memory leak does occur, Netty's leak detection will also be able to identify the issue.